x-pack/plugins/canvas/canvas_plugin_src/functions/common/case.ts

```diff
@@ -5,13 +5,15 @@
  * 2.0.
  */
 
+import { Observable } from 'rxjs';
+import { take } from 'rxjs/operators';
 import { ExpressionFunctionDefinition } from 'src/plugins/expressions/common';
 import { getFunctionHelp } from '../../../i18n';
 
 interface Arguments {
-  when: () => any;
-  if: boolean;
-  then: () => any;
+  when?(): Observable<any>;
```

Comment:
That's because there were non-null checks in the implementation. It's safer to make them optional here to prevent accidental breaking of the implementation. The explicit non-null assertion here is safer from that point because we know that it's defined outside.